dao-contracts/src/reputation.rs

```diff
@@ -134,7 +195,30 @@ impl ReputationContractInterface for ReputationContract {
 
     fn transfer_from(&mut self, owner: Address, recipient: Address, amount: U256) {
         self.access_control.ensure_whitelisted();
-        self.token.raw_transfer(owner, recipient, amount);
+
+        // Load the balance of the owner.
+        let owner_full_balance = self.get_full_balance(owner);
+        let (is_positive_owner, owner_balance) = owner_full_balance;
+
+        // Check if the owner has sufficient balance.
+        if !is_positive_owner || owner_balance < amount {
+            casper_env::revert(Error::InsufficientBalance)
+        }
+
+        // Load the balance of the recipient.
+        let recipient_full_balance = self.get_full_balance(recipient);
+
+        // Settle the transfer.
+        self.balances
+            .set(&owner, rem_from_balance(owner_full_balance, amount));
+        self.balances
+            .set(&recipient, add_to_balance(recipient_full_balance, amount));
+    }
+}
+
+impl ReputationContract {
+    fn get_full_balance(&self, address: Address) -> (bool, U256) {
```

Comment:
I would think of better naming. Maybe `get_signed_balance()`/`get_balanace_with_sign()`.

Reply:
Done.